src/app/project-detail/project-detail.component.ts

```diff
@@ -31,43 +35,47 @@ export class ProjectDetailComponent implements OnInit {
     currentPage: 1
   };
 
-  private sub: Subscription;
+  private subscription: Subscription;
   NewsTypeFilterPipe: NewsTypeFilterPipe;
   NewsHeadlineFilterPipe: NewsHeadlineFilterPipe;
 
-  constructor(private api: Api, private _changeDetectionRef: ChangeDetectorRef, private route: ActivatedRoute, private router: Router) {
+  constructor(private api: Api, private route: ActivatedRoute, private router: Router,
+    private projectService: ProjectService, private newsService: NewsService) {
     this.NewsTypeFilterPipe = new NewsTypeFilterPipe();
     this.NewsHeadlineFilterPipe = new NewsHeadlineFilterPipe();
   }
 
   ngOnInit() {
     this.loading = true;
-    this.sub = this.route.data.subscribe(
-      (data: { project: Project }) => {
-        this.project = new Project(data.project);
-        this.loading = false;
-        // attach host to document url if it goes to esm-server
-        this.setDocumentUrl(this.project);
+    const projectCode = this.route.snapshot.params.code;
 
+    // get project data
+    this.projectService.getByCode(projectCode)
+      .subscribe((data) => {
+        this.project = new Project(data);
         if (!this.project.proponent) {
           this.project.proponent = { name: '' };
         }
         this.column = 'dateAdded';
         this.direction = -1;
-        // Needed in development mode - not required in prod.
-        this._changeDetectionRef.detectChanges();
-      },
-      error => console.log(error)
-    );
+        this.loading = this.loading && this.news ? false : true;
+    });
+
+    // get recent activities
+    this.newsService.getByProjectCode(projectCode)
+      .subscribe((data) => {
+        this.news = data;
+        this.setDocumentUrl();
+        this.loading = this.loading && this.project ? false : true;
+    });
   }
 
-  setDocumentUrl(project) {
+  setDocumentUrl() {
     const regex = /http(s)?:\/\/(www.)?/;
-    project.recent_activities.forEach(activity => {
+    this.news.forEach(activity => {
```

Comment:
I don't really care, but I'd prefer if this function took the 'news' as a parameter.
Makes this function more testable, and I think generally functions that dont do anything on their own (rely on data being set up elsewhere), is less good.

Reply:
Yeah, I don't have a specific preference in this case, but I think you're right. Updating.